src/backend/backend.c

```diff
@@ -31,12 +31,17 @@ struct backend_operations *backend_list[NUM_BKEND] = {
 /**
  * @param all_damage if true ignore damage and repaint the whole screen
  */
-region_t get_damage(session_t *ps, bool all_damage) {
+region_t get_damage(session_t *ps, bool whole_screen) {
 	region_t region;
-	auto buffer_age_fn = ps->backend_data->ops->buffer_age;
-	int buffer_age = buffer_age_fn ? buffer_age_fn(ps->backend_data) : -1;
+	int buffer_age;
+	if (ps->o.legacy_backends) {
+		buffer_age = get_buffer_age(ps);
+	} else {
+		auto buffer_age_fn = ps->backend_data->ops->buffer_age;
+		buffer_age = buffer_age_fn ? buffer_age_fn(ps->backend_data) : -1;
+	}
 
-	if (all_damage) {
+	if (whole_screen) {
 		buffer_age = -1;
 	}
```

Comment:
isn't it better to assume that `buffer_age` is `-1` and call buffer age function only if `whole_screen` is `false`?